packages/core/content-manager/server/src/services/document-manager.ts

```diff
@@ -193,44 +193,27 @@ const documentManager = ({ strapi }: { strapi: Core.Strapi }) => {
       return strapi
         .documents(uid)
         .publish({ ...params, documentId: id })
-        .then((result) => result?.versions.at(0));
+        .then((result) => result?.versions);
+      // TODO __** check this return format change okay?
     },
 
-    async publishMany(entities: Document[], uid: UID.ContentType) {
-      if (!entities.length) {
-        return null;
-      }
-
-      // Validate entities before publishing, throw if invalid
-      await Promise.all(
-        entities.map((document: Document) => {
-          return strapi.entityValidator.validateEntityCreation(
-            strapi.getModel(uid),
-            document,
-            undefined,
-            // @ts-expect-error - FIXME: entity here is unnecessary
-            document
-          );
-        })
-      );
+    async publishMany(uid: UID.ContentType, documentIds: string[], locale?: string | string[]) {
+      let publishedEntitiesCount = 0;
+      await async.map(documentIds, async (documentId: string) => {
+        // Publish per documentId
+        const res = await this.publish(documentId, uid, { locale });
+        publishedEntitiesCount += res?.length ?? 0;
```

Comment:
What happens if one publish failed? I think we throw an error so we can end with only the half of the documentIds published even if we throw an error.

I think we should wrapper this in a transaction.

I think we can save the `publishedEntitiesCount`, because we publish all or we don't publish anything, so at the end we only need to return `documentIds.length`